src/overlays/actors/ovl_En_Bbfall/z_en_bbfall.c

```diff
@@ -37,140 +47,717 @@ const ActorInit En_Bbfall_InitVars = {
+        for (i = 0; i < ARRAY_COUNT(this->bodyPartsPos); i++) {
+            func_800B3030(globalCtx, &this->bodyPartsPos[i], &gZeroVec3f, &gZeroVec3f, 40, 7, 2);
+            SoundSource_PlaySfxAtFixedWorldPos(globalCtx, &this->bodyPartsPos[i], 11, NA_SE_EN_EXTINCT);
+        }
+
+        Actor_MarkForDeath(&this->actor);
+    } else {
+        for (i = 0; i < ARRAY_COUNT(this->bodyPartsPos); i++) {
+            Math_Vec3f_Sum(&this->bodyPartsPos[i], &this->bodyPartsVelocity[i], &this->bodyPartsPos[i]);
+            this->bodyPartsVelocity[i].y += this->actor.gravity;
+        }
+    }
+}
+
+void EnBbfall_SetupDamage(EnBbfall* this) {
+    this->collider.base.acFlags &= ~AC_ON;
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_BUBLE_DAMAGE);
+    func_800BE5CC(&this->actor, &this->collider, 0);
+
+    if (this->actor.colChkInfo.damageEffect == EN_BBFALL_DMGEFF_ZORA_MAGIC) {
+        Actor_SetColorFilter(&this->actor, 0, 255, 0, 40);
+        this->drawDmgEffType = ACTOR_DRAW_DMGEFF_ELECTRIC_SPARKS_LARGE;
+        this->drawDmgEffAlpha = 2.0f;
+        this->drawDmgEffScale = 0.4f;
+    } else if (this->actor.colChkInfo.damageEffect == EN_BBFALL_DMGEFF_STUN) {
+        Actor_SetColorFilter(&this->actor, 0, 255, 0, 20);
+        this->actor.speedXZ = 0.0f;
+    } else if (this->actor.colChkInfo.damageEffect == EN_BBFALL_DMGEFF_HOOKSHOT) {
+        this->actor.speedXZ = 0.0f;
+    } else {
+        Actor_SetColorFilter(&this->actor, 0x4000, 255, 0, 20);
+        this->actor.speedXZ = 7.0f;
+    }
+
+    this->actionFunc = EnBbfall_Damage;
+}
+
+void EnBbfall_Damage(EnBbfall* this, GlobalContext* globalCtx) {
+    Math_SmoothStepToF(&this->actor.speedXZ, 0.0f, 1.0f, 0.5f, 0.0f);
+    if ((this->actor.bgCheckFlags & 1) && (this->actor.speedXZ < 0.1f)) {
+        if (EnBbfall_IsTouchingLava(this, globalCtx)) {
+            EnBbfall_SetupSinkIntoLava(this);
+        } else {
+            EnBbfall_SetupDown(this);
+        }
+    }
+}
+
+void EnBbfall_SetupFrozen(EnBbfall* this) {
+    this->actor.speedXZ = 0.0f;
+    if (this->actor.velocity.y > 0.0f) {
+        this->actor.velocity.y = 0.0f;
+    }
+
+    this->actor.gravity = -2.0f;
+    this->actionFunc = EnBbfall_Frozen;
+}
+
+void EnBbfall_Frozen(EnBbfall* this, GlobalContext* globalCtx) {
+    DECR(this->timer);
+
+    if (this->timer == 0) {
+        EnBbfall_Thaw(this, globalCtx);
+        if (this->actor.colChkInfo.health == 0) {
+            EnBbfall_SetupDead(this, globalCtx);
+        } else {
+            EnBbfall_SetupDown(this);
+        }
+    }
+}
+
+void EnBbfall_UpdateDamage(EnBbfall* this, GlobalContext* globalCtx) {
+    if (this->collider.base.acFlags & AC_HIT) {
+        this->collider.base.acFlags &= ~AC_HIT;
+        this->collider.base.atFlags &= ~(AT_HIT | AT_BOUNCED);
+        this->collider.base.atFlags &= ~AT_ON;
+        if ((this->drawDmgEffType != ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX) ||
+            (!(this->collider.elements[0].info.acHitInfo->toucher.dmgFlags & 0xDB0B3))) {
```

Comment:
```suggestion
            !(this->collider.elements[0].info.acHitInfo->toucher.dmgFlags & 0xDB0B3)) {
```
Personally I don't think it is necessary to add parenthesis when the operation is just `!`, but up to you.

Reply:
I agree. I copied this styling from En_Bb so let me change it there too.